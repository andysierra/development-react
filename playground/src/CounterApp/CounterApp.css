button {
  font-size: 1.5rem;
  padding: 8px 15px 8px 15px;
  outline: none;
  border-color: black;
  border-style: solid;
  border-width: 0px 1px 0px 1px;
}
button:hover {
  background-color: gainsboro;
}
button:active {
  background-color: darkgray;
  box-shadow: inset 2px 2px 2px rgba(100, 100, 100, .5);
}