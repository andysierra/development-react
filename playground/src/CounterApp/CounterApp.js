import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './CounterApp.css';

const CounterApp = ({value}) => {

  const [counter, setCounter] = useState(value);

  const add       = e => setCounter(counter+1);
  const substract = e => setCounter(counter-1);
  const reset     = e => setCounter(value);

  return (
    <div>
      <h1>CounterApp</h1>
      <hr></hr>
      <button onClick={add}>+</button>
      <button onClick={reset}>reset</button>
      <button onClick={substract}>-</button>
      <h2>{counter}</h2>
    </div>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number
}

CounterApp.defaultProps = {
  value: 0
}

export default CounterApp;