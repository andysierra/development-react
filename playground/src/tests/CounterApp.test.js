import CounterApp from "../CounterApp/CounterApp";
import {shallow} from 'enzyme';

describe('Pruebas para el <CounterApp/>', () => {

  let wrapper = shallow(<CounterApp/>);

  beforeEach(()=>{
    wrapper = shallow(<CounterApp/>);
  });

  test('debe de mostrar <CounterApp/> correctamente', () => {
    const value = 10;
    const wrapper = shallow(<CounterApp value={value}/>);
    const resultado = wrapper.find('h2').text();
    //expect(parseInt(resultado)).toBe(value);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el valor por defecto 100', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value}/>);
    const resultado = wrapper.find('h2').text().trim();
    expect(parseInt(resultado)).toBe(value);
  });

  test('debe de incrementar el counter al presionar el botón +1', () => {
    wrapper.find('button').at(0).simulate('click');
    const resultado = wrapper.find('h2').text().trim();
    console.log(resultado);
    expect(parseInt(resultado)).toBe(1);
  });

  test('debe de disminuir el counter al presionar el botón -1', () => {
    wrapper.find('button').at(2).simulate('click');
    const resultado = wrapper.find('h2').text().trim();
    console.log(resultado);
    expect(parseInt(resultado)).toBe(-1);
  });

  test('debe de reiniciar el counter al presionar el botón reset', () => {
    const value = 1993;
    const wrapper = shallow(<CounterApp value={value}/>);
    wrapper.find('button').at(0).simulate('click');
    expect(parseInt(wrapper.find('h2').text().trim())).toBe(value+1);
    wrapper.find('button').at(1).simulate('click');
    expect(parseInt(wrapper.find('h2').text().trim())).toBe(value);
    console.log(wrapper.find('h2').text().trim());
  });
})