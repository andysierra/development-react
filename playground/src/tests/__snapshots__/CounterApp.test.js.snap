// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el <CounterApp/> debe de mostrar <CounterApp/> correctamente 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <h2>
    10
  </h2>
</div>
`;
